packages/token-deposit/scripts/exec.js

```diff
@@ -18,106 +17,111 @@ requireEnvVariables(['DEVNET_PRIVKEY', 'L2RPC', 'L1RPC'])
+   */
+  const expectedL1GatewayAddress = await erc20Bridge.getL1GatewayAddress(erc20Address, l1Provider)
+  const initialBridgeTokenBalance = await l1DappToken.balanceOf(expectedL1GatewayAddress)
 
   /**
    * The Standard Gateway contract will ultimately be making the token transfer call; thus, that's the contract we need to approve.
-   * bridge.approveToken handles this approval
+   * erc20Bridge.approveToken handles this approval
+   * Arguments required are: 
+   * (1) l1Signer: The L1 address transferring token to L2
+   * (2) erc20L1Address: L1 address of the ERC20 token to be depositted to L2
    */
-  const approveTx = await bridge.approveToken(erc20Address)
+  const approveTx = await erc20Bridge.approveToken({
+    l1Signer: l1Wallet,
+    erc20L1Address: erc20Address
+  })
+  
   const approveRec = await approveTx.wait()
   console.log(
     `You successfully allowed the Arbitrum Bridge to spend DappToken ${approveRec.transactionHash}`
   )
 
   /**
-   * Deposit DappToken to L2 using Bridge. This will escrows funds in the Gateway contract on L1, and send a message to mint tokens on L2.
-   * The bridge.deposit method handles computing the necessary fees for automatic-execution of retryable tickets — maxSubmission cost & l2 gas price * gas — and will automatically forward the fees to L2 as callvalue
+   * Deposit DappToken to L2 using erc20Bridge. This will escrow funds in the Gateway contract on L1, and send a message to mint tokens on L2.
+   * The erc20Bridge.deposit method handles computing the necessary fees for automatic-execution of retryable tickets — maxSubmission cost & l2 gas price * gas — and will automatically forward the fees to L2 as callvalue
    * Also note that since this is the first DappToken deposit onto L2, a standard Arb ERC20 contract will automatically be deployed.
+   * Arguments required are: 
+   * (1) amount: The amount of tokens to be transferred to L2
+   * (2) erc20L1Address: L1 address of the ERC20 token to be depositted to L2
+   * (2) l1Signer: The L1 address transferring token to L2
+   * (3) l2Provider: An l2 provider
    */
 
-  const depositTx = await bridge.deposit(erc20Address, tokenDepositAmount)
-  const depositRec = await depositTx.wait()
-
+  const depositTx = await erc20Bridge.deposit({
+    amount: tokenDepositAmount,
+    erc20L1Address: erc20Address,
+    l1Signer: l1Wallet,
+    l2Provider: l2Provider
+  })
+  
   /**
-   * Now we track the status of our retryable ticket
+   * Now we wait for L1 and L2 side of transactions to be confirmed
    */
 
-  //  First, we get our txn's sequence number from the event logs (using a handy utility method). This number uniquely identifies our L1 to L2 message (i.e., our token deposit)
-  const seqNumArr = await bridge.getInboxSeqNumFromContractTransaction(
-    depositRec
-  )
-
+  const depositRec = await depositTx.wait()
+  const message = await depositRec.getL1ToL2Message(l2Provider)
```

Comment:
Here you could instead do `depositRec.waitForL2()`, and check `complete` on the result, and show `.status` of the result.